### PR TITLE
Introduce errorInstanceId for errors telemetry

### PR DESCRIPTION
**Pull request by markfields**
This is a scoped version of #6968.

* Add `errorInstanceId` to `IFluidErrorBase` (which none of our error classes implement yet)
* In `wrapError`, add the inner error's `errorInstanceId` to the wrapping error if present
* Add `wrapErrorandLog` which wraps then logs the inner error.
* No longer copy telemetry props from inner error in `wrapError`, since we can now tie the inner/outer errors together in telemetry via ErrorInstanceId.

### CODE REVIEW

**Comment by markfields** [comment]
@jvoels - Consider if you'd like us to backport this to 0.45 once it goes in, to get into Office Fluid stuff sooner since you're looking forward to this change.  cc @jatgarg

File: packages/loader/container-utils/src/error.ts [hunk 1]
```diff
@@ -149,28 +149,55 @@ export const extractSafePropertiesFromMessage = (message: ISequencedDocumentMess
...
+        newError.addTelemetryProperties({ innerErrorInstanceId: innerError.errorInstanceId });
     }
 
     return newError;
 }
+
+/** The same as wrapError, but also logs the innerError, including the wrapping error's instance id */
+export function wrapErrorAndLog<T extends IWriteableLoggingError>(
+    innerError: unknown,
+    newErrorFn: (mmessage: string) => T,
```
Comment (jatgarg) [inline packages/loader/container-utils/src/error.ts:189]: nit: message

**Review by jatgarg — APPROVED** [review]
:shipit:

File: packages/loader/container-utils/src/error.ts (hunk 1 above)
Comment (anthony-murphy) [reply packages/loader/container-utils/src/error.ts]: fixed this

**Pull request merged**